rediscluster/nodemanager.py

```diff
@@ -65,7 +67,7 @@ def random_node(self):
         return self.nodes[key]
 
     def get_redis_link(self, host, port, decode_responses=False):
-        return StrictRedis(host=host, port=port, decode_responses=decode_responses)
+        return StrictRedis(host=host, port=port, decode_responses=decode_responses, **self.connection_kwargs)
```

Comment:
On Fri, Jul 17, 2015 at 3:07 PM, Grok <notifications@github.com> wrote:

> In rediscluster/nodemanager.py
> <https://github.com/Grokzen/redis-py-cluster/pull/78#discussion_r34931192>
> :
>
> > @@ -65,7 +67,7 @@ def random_node(self):
> >          return self.nodes[key]
> >
> >      def get_redis_link(self, host, port, decode_responses=False):
> > -        return StrictRedis(host=host, port=port, decode_responses=decode_responses)
> > +        return StrictRedis(host=host, port=port, decode_responses=decode_responses, **self.connection_kwargs)
>
> The problem with your tests is here. The connection args that is sent in
> test_repr_contains_db_info_unix test should not be sent down to
> StrictRedis i guess. The connection_args that goes into
> ClusterConnectionPool is not the same as the args that is going into
> ClusterConnection class, there must be some way to seperate them.
>
Yeah, I had fixed this in my branch (see
https://github.com/dfb/redis-py-cluster/commit/e3758258bd67ee47f8df0cea9bcfbb64d051d64e),
but then found other places where the tests are failing due to kwargs - for
example, one test passes 'foo' and 'bar' kwargs.

I'm going to pause working on my patch for awhile and come back to it if I
can think of a clean way to handle this.